packages/frontend/src/shared/remark/remark-iex.ts

```diff
@@ -33,153 +31,209 @@ export const remarkIex = (options) => {
       case 'badge':
-        node.type = 'badge';
+        node.data = {
+          hName: 'badge',
+          hProperties: node.attributes
+        };
         break;
-      case 'image': {
-        node.type = 'image';
-        node.url = node.children[0]?.value ?? node.children[0]?.url;
-        delete node.children;
 
+      case 'image': {
         const { height: h, width: w, ...attributes } = node.attributes;
 
-        node.attributes = {
-          w,
-          h,
-          ...attributes
+        node.data = {
+          hName: 'img',
+          hProperties: {
+            src: node.children[0]?.value ?? node.children[0]?.url,
+            w,
+            h,
+            ...attributes
+          }
         };
 
+        delete node.children;
         break;
       }
+
       case 'video': {
-        node.type = 'video';
-        node.url = node.children[0]?.value ?? node.children[0]?.url;
-        delete node.children;
+        node.data = {
+          hName: 'video',
+          hProperties: {
+            src: node.children[0]?.value ?? node.children[0]?.url,
+            ...node.attributes
+          }
+        };
 
-        const { autoplay, loop, ...attributes } = node.attributes;
+        delete node.children;
+        break;
+      }
 
-        node.attributes = {
-          // Convert from string to boolean
-          autoPlay: autoplay === 'true',
-          loop: loop === 'true',
-          ...attributes
+      case 'katex': {
+        node.data = {
+          hName: 'katex',
+          hProperties: {
+            math: remarkNodetoText(node)
+          }
         };
 
         break;
       }
-      case 'katex':
-        node.type = 'inlineMath';
-        node.value = remarkNodetoText(node);
-        break;
+
+      default:
+        node.data = {
+          hName: 'textdirective',
+          hProperties: node
+        };
     }
   }
 
   function leafDirectives(node) {
     switch (node.name) {
-      case 'insight':
-        node.type = 'insight';
-        if (node.children.length > 0) {
-          node.fullName = node.children[0].value;
-        }
+      case 'insight': {
+        node.data = {
+          hName: 'insight',
+          hProperties: {
+            ...node.attributes,
+            fullName: remarkNodetoText(node)
+          }
+        };
+
         break;
-      case 'insights':
-        node.type = 'insights';
-        if (node.children.length > 0) {
-          node.query = node.children[0].value;
-        }
+      }
+
+      case 'insights': {
+        node.data = {
+          hName: 'insights',
+          hProperties: {
+            ...node.attributes,
+            query: remarkNodetoText(node)
+          }
+        };
+
         break;
+      }
+
+      default:
+        node.data = {
+          hName: 'leafdirective',
+          hProperties: node
+        };
     }
   }
 
   function containerDirectives(node) {
     switch (node.name) {
+      case 'table':
       case 'custom_table': {
```

Comment:
Backwards compatible with `:::custom_table`, including converting the border/no-border into an attribute